tools/dotcms-cli/cli/src/main/java/com/dotcms/api/client/pull/file/FileFetcher.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    /**
+     * Retrieves file traversal result based on the provided path and custom options.
+     *
+     * @param path          The path to traverse.
+     * @param customOptions The custom options for traversal.
+     * @return The file traversal result.
+     */
+    private FileTraverseResult fetch(String path, Map<String, Object> customOptions) {
+
+        if (LocationUtils.isFolderURL(path)) { // Handling folders
+
+            Set<String> includeFolderPatterns = getSetFromOptions(customOptions,
+                    INCLUDE_FOLDER_PATTERNS);
+            Set<String> includeAssetPatterns = getSetFromOptions(customOptions,
+                    INCLUDE_ASSET_PATTERNS);
+            Set<String> excludeFolderPatterns = getSetFromOptions(customOptions,
+                    EXCLUDE_FOLDER_PATTERNS);
+            Set<String> excludeAssetPatterns = getSetFromOptions(customOptions,
+                    EXCLUDE_ASSET_PATTERNS);
+
+            boolean nonRecursive = false;
+            if (customOptions != null) {
+                nonRecursive = (boolean) customOptions.getOrDefault(NON_RECURSIVE, false);
+            }
+
+            // Service to handle the traversal of the folder
+            var response = remoteTraversalService.traverseRemoteFolder(
+                    path,
+                    nonRecursive ? 0 : null,
+                    true,
+                    includeFolderPatterns,
+                    includeAssetPatterns,
+                    excludeFolderPatterns,
+                    excludeAssetPatterns
+            );
+
+            return FileTraverseResult.builder().
+                    tree(response.getRight()).
+                    exceptions(response.getLeft()).
+                    build();
+
+        } else { // Handling single files
+
+            var asset = retrieveAssetInformation(path);
+            return FileTraverseResult.builder().asset(asset).build();
+        }
+    }
+
+    /**
+     * Retrieves the asset information
+     *
+     * @param source The asset path
+     * @return The asset information
+     */
+    AssetVersionsView retrieveAssetInformation(final String source) {
+
+        // Requesting the file info
+        final AssetAPI assetAPI = this.clientFactory.getClient(AssetAPI.class);
+
+        // Execute the REST call to retrieve asset information
+        String encodedURL = encodePath(source);
+        var response = assetAPI.assetByPath(ByPathRequest.builder().assetPath(encodedURL).build());
+        return response.entity();
+    }
+
+    /**
+     * Retrieves the set of values associated with the given key in the provided options map.
+     *
+     * @param options The map of options
+     * @param key     The key to retrieve the values for
+     * @return The set of values associated with the key, or null if the options map is null or does
+     * not contain the key
+     */
+    private Set<String> getSetFromOptions(Map<String, Object> options, String key) {
```

Comment:
set to final

Reply:
this could be reduced to something such as

return Optiona.ofNullable(options). ifPresentOrElse(map -> options.getOrDefault(key, Collections.emptySet(), ()->Collections.emptySet());